sdk/cosmos/azure-cosmos/azure/cosmos/_auth_policies.py

```diff
@@ -0,0 +1,161 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+import time
+from typing import Any, Dict, Optional
+from azure.core.credentials import AccessToken, TokenCredential
+from azure.core.pipeline import PipelineRequest, PipelineResponse
+from azure.core.pipeline.policies import HTTPPolicy
+
+
+# pylint:disable=too-few-public-methods
+class _CosmosBearerTokenCredentialPolicyBase(object):
+    """Base class for a Bearer Token Credential Policy.
+
+    :param credential: The credential.
+    :type credential: ~azure.core.credentials.TokenCredential
+    :param str scopes: Lets you specify the type of access needed.
+    """
+
+    def __init__(self, credential, *scopes, **kwargs):  # pylint:disable=unused-argument
+        # type: (TokenCredential, *str, **Any) -> None
+        super(_CosmosBearerTokenCredentialPolicyBase, self).__init__()
```

Comment:
Why are we calling `super` when we're not inheriting from anything?
Though having said that - given that 90% of this is identical to the Core implementation - is there any reason you're not inheriting from the Core policy and simply overriding `_update_headers`?

Reply:
Yeah good point. And these are technically internal if we swap out the implementation later.